virtinst/virtxml.py

```diff
@@ -252,21 +350,25 @@ def start_domain_transient(conn, xmlobj, devs, action, confirm):
 
 
 def update_changes(domain, devs, action, confirm):
-    if action == "hotplug":
+    if action.is_add_device:
         msg_confirm = _("%(xml)s\n\nHotplug this device to the guest "
                         "'%(domain)s'?")
         msg_success = _("Device hotplug successful.")
         msg_fail = _("Error attempting device hotplug: %(error)s")
-    elif action == "hotunplug":
+    elif action.is_remove_device:
         msg_confirm = _("%(xml)s\n\nHotunplug this device from the guest "
                         "'%(domain)s'?")
         msg_success = _("Device hotunplug successful.")
         msg_fail = _("Error attempting device hotunplug: %(error)s")
-    elif action == "update":
+    elif action.is_edit:
         msg_confirm = _("%(xml)s\n\nUpdate this device for the guest "
                         "'%(domain)s'?")
         msg_success = _("Device update successful.")
         msg_fail = _("Error attempting device update: %(error)s")
+    else:
+        raise xmlutil.DevError(
+                "update_changes() incorrectly called with action=%s" %
+                action.action_name)
```

Comment:
`action_string`

Reply:
argh, good catch. I'll fix by using `action_name` everywhere since there's already a mix of it and `action_string`